### ISSUE TITLE
maml.clone() causes grads to become null

### ISSUE THREAD

**Issue by vkkhare** [learnables/learn2learn#152]
```python
learner = maml.clone()
loss = learner.compute_loss(batch)
diff_params = [p for p in learner.parameters() if p.requires_grad]
grad_params = grad(loss,
                                 diff_params,
                                 retain_graph=True,
                                 create_graph=True,
                                 allow_unused=True)
```

Can someone let me know why do I get null for grads when I do clone while everything works perfectly fine if I adapt without cloning?

my Maml module is
`l2l.algorithms.MAML(mini_model, lr=config["optimizer"]["lr"], first_order=False, allow_nograd=True)`

**Comment by seba-1511** [comment]
Hi @vkkhare ,

Does `learner.compute_loss()` compute the MAML loss (i.e. it adapts the learner) or is it simply compute the cross-entropy loss on that batch?

When you say the gradients are null, do you mean you get `None` back or do you get zero tensors?

**Comment by vkkhare** [comment]
Thanks for the quick response @seba-1511! learner.compute_loss only evaluates the cross_entropy loss. I have to explicitly call backward on this to set the gradients

**Comment by seba-1511** [comment]
Thanks, that's perplexing. Would it be possible to have a minimal reproducible example, maybe hosted on colab? It's difficult to tell you what's happening without having the implementation for `compute_loss`.

**Comment by vkkhare** [comment]
Okay I will do it soon

**Comment by seba-1511** [comment]
Inactive.

**Comment by hcyoo93** [comment]
Hello, @vkkhare

Did you resolve this issue?
I am encountering the same issue... 
If you did, would you please share your experience?

It would be very helpful for many
